src/tools/starlark/java/com/google/devtools/starlark/common/DocstringUtils.java

```diff
@@ -69,6 +69,63 @@ public static DocstringInfo parseDocstring(String doc, List<DocstringParseError>
     return result;
   }
 
+  /**
+   * Removes leading whitespace from docstring lines after the first.
+   *
+   * Allows docstrings to be indented to the context where they appear, without including that indent in the output.
+   * Some output formats have semantic meaning for leading whitespace (such as Markdown).
+   *
+   * @param doc A non-function docstring, without Args or Returns sections.
+   * 
+   */
+  public static String dedentDocstring(String doc) {
+    // Two iterations: first buffer lines and find the minimal indent, then trim to the min
+    List<String> buffer = new ArrayList<>();
+    int indentation = Integer.MAX_VALUE;
+    int lineOffset = 0;
+    int endOfLineOffset = 0;
+
+    do {
+      endOfLineOffset = doc.indexOf("\n", lineOffset);
+      String line = endOfLineOffset < 0 ? doc.substring(lineOffset) : doc.substring(lineOffset, endOfLineOffset + 1);
+      boolean allWhitespace = line.trim().isEmpty();
+      boolean isFirstLine = lineOffset == 0;
+      if (!allWhitespace) {
+        int lineIndent = DocstringParser.getIndentation(line);
+        if (isFirstLine && lineIndent == 0) {
+          // Doesn't count. First line is often not indented, e.g.
+          // doc = """First line
+          //       second line"""
+          // should still result in both lines with no leading whitespace
+        } else {
+          indentation = Math.min(indentation, lineIndent);
+        }
+      }
+      buffer.add(line);
+      // next line
+      lineOffset = endOfLineOffset + 1;
+    } while (endOfLineOffset >= 0);
+
+    if (indentation == Integer.MAX_VALUE) {
+      return doc;
+    }
+
+    boolean firstLine = true;
+    StringBuilder description = new StringBuilder();
+    for (String bufLine : buffer) {
+      if (firstLine) {
+        description.append(bufLine);
+        firstLine = false;
+      } else if (bufLine.trim().isEmpty()) {
+        description.append("\n");
```

Comment:
Returning `doc` unchanged on line 110 implies that all-whitespace lines become empty lines (even if they contain whitespace beyond `indentation`) if any dedentation is done, but remain unchanged when not dedenting - which is unintuitive behavior.

Would it make more sense to instead dedent all-whitespace lines by up to the same number of spaces as any other line?